### ISSUE TITLE
failed to call model.forward() of faster rcnn

### ISSUE THREAD

**Issue by KouWeiBin** [pytorch/vision#4093]
## 🐛 Bug
```
terminate called after throwing an instance of 'std::runtime_error'
  what():  The following operation failed in the TorchScript interpreter.
Traceback of TorchScript, serialized code (most recent call last):
  File "code/__torch__/torchvision/models/detection/faster_rcnn.py", line 1799, in forward
        _618 = torch.slice(offsets, 0, 0, 9223372036854775807, 1)
        boxes_for_nms = torch.add(boxes5, torch.unsqueeze(_618, 1), alpha=1)
        keep2 = ops.torchvision.nms(boxes_for_nms, scores0, _611)
                ~~~~~~~~~~~~~~~~~~~ <--- HERE
        keep1 = keep2
      if _417.training:

Traceback of TorchScript, original code (most recent call last):
  File "/home/kevin/anaconda3/envs/script/lib/python3.6/site-packages/torchvision/ops/boxes.py", line 40, in forward
        by NMS, sorted in decreasing order of scores
    """
    return torch.ops.torchvision.nms(boxes, scores, iou_threshold)
           ~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
RuntimeError: Could not run 'torchvision::nms' with arguments from the 'CUDA' backend. 'torchvision::nms' is only available for these backends: [CPU].
```

## To Reproduce

1. train model using faster rcnn
2. convert the model to script model
3. load he model in qt successfully
4. it occurs a error when call function model.forward(inputs)

```c
    cv::Mat pred_img = m_predict_img.clone();
#ifdef RGB
    pred_img.convertTo(*m_img_float, CV_32FC3, 1.0 / 255.0);
#else
    pred_img.convertTo(*m_img_float, CV_32FC1, 1.0 / 255.0);
#endif
    std::cout << "mark1" << std::endl;
    torch::Tensor tensor_image = torch::from_blob(m_img_float->data, {m_img_float->rows,
                                                  m_img_float->cols, m_img_float->channels()}, torch::kF32);
    tensor_image = tensor_image.permute(/*{0, 3, 1, 2}*/{2, 0, 1});
    auto img_var = torch::autograd::make_variable(tensor_image, false);

    std::cout << "mark2" << std::endl;
    std::vector<torch::jit::IValue> inputs;
    torch::jit::IValue output;
    if (check_gpu_available()) {
        inputs.emplace_back(c10::List<at::Tensor>({img_var.to(torch::kCUDA)}));
        std::cout << "mark3" << std::endl;
        output = m_crop_module.forward(inputs);
        std::cout << "mark4" << std::endl;
    } else {
        inputs.emplace_back(c10::List<at::Tensor>({img_var.to(torch::kCPU)}));
        output = m_crop_module.forward(inputs);
    }
    inputs.pop_back();
```

## Expected behavior

can work well

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
ubuntu20.4+libtorch1.6+torchvision0.7
```

**Comment by fmassa** [comment]
Hi,

I think you might have installed the CPU version of torchvision when running it in C++, can you double-check?

**Comment by KouWeiBin** [comment]
> Hi,
> 
> I think you might have installed the CPU version of torchvision when running it in C++, can you double-check?

actually I have checked it and its .so list can be checked as follow:
(script) kevin@wbkou:/usr/local/torchvision/lib$ ldd libtorchvision.so
	linux-vdso.so.1 (0x00007ffdfb5c3000)
	libtorch.so => /usr/local/libtorch/lib/libtorch.so (0x00007f31f2b79000)
	libc10.so => /usr/local/libtorch/lib/libc10.so (0x00007f31f2901000)
	libcuda.so.1 => /lib/x86_64-linux-gnu/libcuda.so.1 (0x00007f31f11fe000)
	libnvrtc.so.10.1 => /usr/local/libtorch/lib/libnvrtc.so.10.1 (0x00007f31efa4f000)
	libnvToolsExt.so.1 => /usr/local/libtorch/lib/libnvToolsExt.so.1 (0x00007f31ef845000)
	libcudart.so.10.1 => /lib/x86_64-linux-gnu/libcudart.so.10.1 (0x00007f31ef5c9000)
	libc10_cuda.so => /usr/local/libtorch/lib/libc10_cuda.so (0x00007f31ef38f000)
	libpython3.6m.so.1.0 => /home/kevin/anaconda3/envs/script/lib/libpython3.6m.so.1.0 (0x00007f31ef048000)
	libtorch_cpu.so => /usr/local/libtorch/lib/libtorch_cpu.so (0x00007f31deb22000)
	libtorch_cuda.so => /usr/local/libtorch/lib/libtorch_cuda.so (0x00007f319fb25000)
	libstdc++.so.6 => /home/kevin/anaconda3/envs/script/lib/libstdc++.so.6 (0x00007f319f9b0000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f319f861000)
	libgcc_s.so.1 => /home/kevin/anaconda3/envs/script/lib/libgcc_s.so.1 (0x00007f319f84b000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f319f659000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f31f2f91000)
	libgomp-75eea7e8.so.1 => /usr/local/libtorch/lib/libgomp-75eea7e8.so.1 (0x00007f319f434000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f319f411000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f319f40b000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f319f400000)
	libcudart-80664282.so.10.2 => /usr/local/libtorch/lib/libcudart-80664282.so.10.2 (0x00007f319f17d000)
	libutil.so.1 => /lib/x86_64-linux-gnu/libutil.so.1 (0x00007f319f178000)
	libtensorpipe.so => /usr/local/libtorch/lib/libtensorpipe.so (0x00007f319ec7a000)

**Comment by fmassa** [comment]
Hi,

Sorry for the delay in getting back to you. Is this still an issue for you?

**Comment by KouWeiBin** [comment]
> Hi,
> 
> Sorry for the delay in getting back to you. Is this still an issue for you?

Thanks for your reply!
It has been solved by updating lib torch to 1.8.0 and torchvision to 0.9.
